Initializing percy testing

immortalcodes opened the pull request:
## Rationale
- For now the plan is to set up a basic mechanism that could be triggered on PRs merging into main, this mechanism will take and test snapshots and highlight the differences and then devs will have to approve the new snapshots. 
- On approval these screenshots will form the base build to compare the screenshots on next PR, this will go in a way that our base build is always updated with latest changes.
- Idea is not to include all pages, just some main pages/bubbles and we are yet to decide on that.
- Now this is still very initial phase and we are likely to encounter more questions and edge cases, we can refine our approach for them as we move forward.

## Done
- Added percy package.
- Create config file for snapshots.
- Created github action to take snapshots.
- Created workflow to run percy action on PRs targeting main on every pushed commit.
- Created the repo on percy site to accept and compare builds.

## QA

- Check out this feature branch
-  To run it locally just run dotrun in one terminal and then run `dotrun percy`.
- Note: You would need a token for this, please ask the author for same.
- To test on PRs, just create a branch of this branch, change something, make a PR and see if percy tests are being run.
- See the builds on https://percy.io/bb49709b/web/ubuntu.com-eaf28f99 .

## Issue / Card

Fixes # [WD-20582](https://warthogs.atlassian.net/browse/WD-20582)

## Screenshots

[If relevant, please include a screenshot.]


## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)


[WD-20582]: https://warthogs.atlassian.net/browse/WD-20582?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://ubuntu-com-14899.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/7410)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

mtruj013 reviewed (commented):
Thanks @immortalcodes. Code LGTM, but there's some weirdness in the qa. `dotrun percy` ran successfully, I can see the build in the dashboard (#14, test1), and action does run on the [pr](https://github.com/canonical/ubuntu.com/pull/14916). But the actual page I changed isn't what shows up in the build (I changed /ai and it shows /download/desktop) and the action fails. Are either of these expected?

immortalcodes commented:
> Thanks @immortalcodes. Code LGTM, but there's some weirdness in the qa. `dotrun percy` ran successfully, I can see the build in the dashboard (#14, test1), and action does run on the [pr](https://github.com/canonical/ubuntu.com/pull/14916). But the actual page I changed isn't what shows up in the build (I changed /ai and it shows /download/desktop) and the action fails. Are either of these expected?

For now I have just added 2 pages to config, homepage and desktop/download, you would have to add pages to config to see their screenshots in percy.
The last test failed because GTM failed to load
![image](https://github.com/user-attachments/assets/54c541fc-29ff-4fcd-a03d-5e42e5c82272)
I guess I will have to increase the timeout.

mtruj013 commented:
Thanks for clarifying @immortalcodes. I've made some more test commits targeting /download/desktop but it looks like the [action](https://github.com/canonical/ubuntu.com/actions/runs/14180428389/job/39725088364?pr=14916) is still failing for the same reason, could you have a look?

mtruj013 reviewed (approved):
Thanks @immortalcodes, LGTM
